crowd_evacuation/civilian_agent.py

```diff
@@ -70,15 +95,41 @@ def step(self):
         path = path_finding.find_path(self.model.graph, self.pos, self._goal)
         # self._take_shortest_path(possible_steps)
         if path is not None:
-            if self.model.grid.is_cell_empty(path[1]):
-                self.model.grid.move_agent(self, path[1])
+            if len(path) <= self._speed:
+                self.decide_move_action(len(path), path)
+            else:
+                # truncated the path according to the speed of the agent
+                del(path[self._speed+1:])
+                self.decide_move_action(self._speed + 1, path)
+
```

Comment:
I think putting this if else inside the decide_move_action would make a cleaner step() function.